src/node/desktop/src/main/detect_r.ts

```diff
@@ -37,116 +42,98 @@ function showRNotFoundError(msg: string): void {
  * @returns true if startup should continue, false on fatal error
  */
 export async function prepareEnvironment(): Promise<boolean> {
-  if (process.platform === 'win32') {
-    return await prepareEnvironmentWin32();
-  } else {
-    return await prepareEnvironmentPosix();
-  }
-}
 
-async function prepareEnvironmentPosix(): Promise<boolean> {
-
-  assert((process.platform !== 'win32'));
-
-  // check for which R override
-  let rWhichRPath = new FilePath();
-  const whichROverride = getenv('RSTUDIO_WHICH_R');
-  if (whichROverride) {
-    rWhichRPath = new FilePath(whichROverride);
+  try {
+    return prepareEnvironmentImpl();
+  } catch (error) {
+    logger().logError(error);
+    return false;
   }
 
-  let rLdScriptPath = new FilePath();
-  if (process.platform === 'darwin') {
-    if (app.isPackaged) {
-      rLdScriptPath = appState().scriptsPath?.completePath('session/r-ldpath') ?? new FilePath();
-      if (!rLdScriptPath.existsSync()) {
-        rLdScriptPath = new FilePath(app.getAppPath()).completePath('r-ldpath');
-      }
-    } else {
-      rLdScriptPath = appState().sessionPath?.completePath('../r-ldpath') ?? new FilePath();
-    }
-  } else {
-    // determine rLdPaths script location
-    rLdScriptPath = appState().supportingFilePath().completePath('bin/r-ldpath');
-    if (!rLdScriptPath.existsSync()) {
-      rLdScriptPath = appState().supportingFilePath().completePath('session/r-ldpath');
-    }
-  }
+}
+
+async function prepareEnvironmentImpl(): Promise<boolean> {
 
   // attempt to detect R environment
-  const detectResult = await detectREnvironment(rWhichRPath, rLdScriptPath, '');
-  if (!detectResult.success) {
-    showRNotFoundError(detectResult.errMsg ?? 'Unknown error');
+  let rEnvironment : REnvironment;
+  try {
+    rEnvironment = await detectREnvironment();
+  } catch (error) {
+    showRNotFoundError(error ?? 'An unknown error occurred.');
```

Comment:
In general shouldn't need to worry about this (`?? 'An unknown error occurred.`). I was using the pattern at one point but it seems unnecessary. We should only ever throw `Error` or subclasses of such in our code, and it is safe to assume any code we consume is doing the same (including node and Javascript itself).